source/common/http/filter_manager.h

```diff
@@ -674,7 +674,11 @@ class FilterManager : public ScopeTrackedObject,
   }
 
   void addAccessLogHandler(AccessLog::InstanceSharedPtr handler) {
-    access_log_handlers_.push_back(std::move(handler));
+    if (std::dynamic_pointer_cast<AccessLog::FilterPtr>(handler)) {
+      access_log_handlers_.push_front(std::move(handler));
+    } else {
+      access_log_handlers_.push_back(std::move(handler));
+    }
```

Comment:
See, this change still change the order of filter loggers and make it mismatch with order of `http_filters`. And it's hard to say this is a robust way to do this.

And I don't know why dynamic casting  to `AccessLog::FilterPtr` help. See, the access loggers from configuration needn't be added to the `access_log_handlers_`. You can just to extend `FilterManagerCallbacks` to use a new method to return the loggers from configuration.

And when doing the `log()`, you can evaluate the `access_log_handlers_` first. And evaluate the loggers from configuration.
Of course, runtime guard is necessary for this behavior chagne.

/wait